corehq/apps/performance_sms/forms.py

```diff
@@ -166,6 +166,9 @@ def __init__(self, domain, config, *args, **kwargs):
             *form_layout
         )
 
+    def clean_schedule(self):
+        return ScheduleConfiguration.wrap(self.cleaned_data['schedule'])
```

Comment:
Ignore me if I'm way off, but it seems that the `schedule` field in _advanced mode_ is a different format and has a different meaning than `schedule` in _simple mode_: in _advanced_ it's a dict, while in _simple_ it's a string (I think). This makes it harder to reason about the data in an abstract way and could be a source of subtle bugs if the two ever get confused.

Suggestion: the _simple mode_ field should be named something like `schedule_interval` while the _advanced mode_ field can keep the name `schedule`.